### PR TITLE
[Fixes #4218] Removes ServiceLocator pattern from RazorView

### LINKED ISSUE #4218: ServiceLocator: Fix RazorView.RenderAsync
Take the service in the ctor

```
            _bufferScope = context.HttpContext.RequestServices.GetRequiredService<IViewBufferScope>();
```

### PR DESCRIPTION
**Pull request by javiercn**

### CODE REVIEW

File: src/Microsoft.AspNetCore.Mvc.Razor/RazorViewEngine.cs [hunk 1]
```diff
@@ -49,7 +51,8 @@ public class RazorViewEngine : IRazorViewEngine
             IRazorPageActivator pageActivator,
             HtmlEncoder htmlEncoder,
             IOptions<RazorViewEngineOptions> optionsAccessor,
-            ILoggerFactory loggerFactory)
+            ILoggerFactory loggerFactory,
+            IViewBufferScope bufferScope)
```
Comment (rynowak) [inline src/Microsoft.AspNetCore.Mvc.Razor/RazorViewEngine.cs]: This won't work, `IViewBufferScope` is scoped

**Comment by javiercn** [comment]
@rynowak Updated

**Comment by rynowak** [comment]
:shipit:

**Comment by NTaylorMullen** [comment]
:shipit:

**Pull request merged**